### ISSUE TITLE
Reminder emails for users who have started a story but haven't published, haven't added tags, etc.

### ISSUE THREAD

**Issue by jwirfs-brock** [denverfoundation/storybase#580]
TODO:
- [x] Finalized list of reminders
- [x] Decide how often these will be run
- [x] Decide how the background task will be launched
- [x] Copy for e-mails

**Comment by jwirfs-brock** [comment]
We could prompt users to:
- publish stories they have started but not finished
- add tags and meta-data to existing stories
- share the stories they've published
- turn their story into a connected story

...others?

**Comment by jwirfs-brock** [comment]
Just adding some brainstorming about the email reminders...

**Reminder: Publish stories they have started but not finished**
- When to send: Should have some time limit (i.e., it's been in draft mode for 48 hours)
- Content: Should include some helpful tips/links ("Are you stuck...?")
- Questions: Do we send multiple follow-up emails, or just one?

**Reminder: Add tags and meta-data to existing stories**
- When to send: Right after story has been published
- Content: How to tag, how to add locations, how to set a featured image
- Question: This could be part of a "Congratulations, your story is published, now what?" email that includes prompts for sharing the story, embedding it, etc. (see below)

**Reminder: Now that you've published a story, share/embed it**
- When to send: Right after story has been published; again, 24 or 48 hours later
- Content: Link to story detail page, instructions for embedding, instructions for sharing via social media
- Question: Could be in same email as above? Or should these be separate?

**Reminder: Turn your story into a connected story**
- Question: Is this a good idea? Would this be annoying?

**Story-stats: Send out page views after the story has been published**
Question: Would users want to see some of the information we collect through Google Analytics? (Page views, time on page, etc.) What about number of times the story has been shared via Add This?

@ghing I'm just throwing some stuff out here. I'm not sure what is possible technically. Are these ideas doable?

@bpawlak What is your take on the type of reminders that would be most useful?

@tekhneco Any thoughts from a customer development perspective, in terms of encouraging long-term engagement without being too naggy?

**Comment by bpawlak** [comment]
I think that's the biggest issue... not being too "naggy."  We definitely have to come up with some rules about when we email a user about unpublished stories - and how often.  

Maybe after a story has been left unpublished for a month, we send an email reminder saying "you've got # total number of unpublished stories" (and yes, add some helper text on if they're stuck, etc.).  But then we use that total # of stories to consider them all to have been "reminded."  It's only when a user creates a _new_ story AND it goes unpublished for a month that a new reminder would be emailed.   That way users would - at most - get a reminder once-a-month about unpublished stories.  Assuming, that is, that they're creating a new story, once a month, after the email goes out _and_ that new story goes unpublished for a month.  

I think the "add tags" message should _only_ be part of the on-screen instructions after a story is published (I don't think it's necessary to include in an email).  

The "Reminder: Now that you've published a story, share/embed it" email should go out as soon as the story is published - and only once.

**Comment by ghing** [comment]
@jwirfs-brock, to answer your "is this doable" question, the only tricky one is "Reminder: Publish stories they have started but not finished" because sending a message once and only once is more complicated because you have to track stories-to-notifcations.

**Comment by jwirfs-brock** [comment]
@ghing Thanks, that's good to know. That is actually the most important reminder, I think. We have a lot of stories that people have worked on that are sitting in draft mode limbo. Any suggestions for an easy way to do this? Is the suggestion @bpawlak proposed, about doing a monthly "# of unpublished stories" reminder any easier?

**Comment by ghing** [comment]
@jwirfs-brock, it's the same crux.  The hard part is the combination of sending a notification later (in a month) and only once.  I can figure out a good way to do this.

**Comment by wendynorris** [comment]
Sorry Gmail has been futzed all morning and I couldn't see your note, Jordan. 

Actually from a workflow perspective, I'd recommend sending the draft reminder alert 5 days after the story was last edited (that may be different than the creation date). A month later the story is probably stale and there's good reason it's been abandoned. 

I really like the idea of adding a few helpful prompts about common writing blocks to help the author get unstuck. The trick is to avoid sounding like the universally reviled Microsoft Clippy. Easier said than done. 

Instead of sending a post-publication email about meta data, sharing, etc., could we build in a new screen prompt at the end of the builder process? Or rework the account page and add a link to a new page with tips/hints about expanding story reach via social media, embedding and meta data? 

I think anything else beyond these two primary needs is going to quickly begin to feel pretty spammy. 

The Google Analytics issue, in my mind, is something very different than above and a crucial element for long term engagement. Right now, authors don't have any metric of story reach or impact. Therefore, they're nto getting the feedback they need to continue publishing. If we can parse some of the story-specific analytics that would also be a nice addition to the account page.

**Comment by jwirfs-brock** [comment]
@bpawlak @tekhneco Those are good suggestions about the function of the reminder. I think the most important thing is to nudge people to finish unpublished stories that have sat dormant.

So, for now, let's do:
- **Email remding users to finish an unpublished story**, sent out five days after the last time the story was edited. I think it's best to only send out one reminder email per story. (That is, if the user edits the story again, then it sits dormant for five more days, we don't send out a second reminder.)
- **Email immediately after a story has been published that contains the story link and suggestions for sharing the story**. We might even want to include the thumbnail and story summary directly in the body of the email.

I agree with Wendy and Bill's suggestions that the information about tagging, sharing, embedding, etc. should go elsewhere (accounts page, published viewed of publish stage in builder). But I do think it will be handy to send users an email confirmation for published stories. That way, they can forward the email along as one simple way of sharing the story.

I'll open another issue about adding friendly reminders elsewhere on the site.

**Comment by wendynorris** [comment]
Sounds like a good way forward.

We should make a note to revisit the reminder/nudge process:

1) after implementing it to get some user feedback on its effectiveness

2) to monitor Google Analytics for changes in baseline visits/views/shares
and

3) to bake any refinements directly into the Story Builder 2.0 user story
creation/publication process.

**
_wendy norris http://about.me/wendynorris_
ceo + founder :: tekhne media, inc
a digital information and media collective
720 281 9378 :: mobile

*\* http://about.me/wendynorris
*
*

On Mon, Jul 1, 2013 at 6:56 AM, Jordan Wirfs-Brock <notifications@github.com

**Comment by jwirfs-brock** [comment]
Final list of email reminders:
1. Unpublished story reminder
   To be sent: 5 days after last edit for unpublished stories; only gets sent once per story
   Draft copy: https://docs.google.com/document/d/1keJOi0JdSfT8LfA674hVl64Uz15llF48N8O-h3noeuQ/edit
2. Published story email
   To be send: Immediately after a new story is published
   Draft copy: https://docs.google.com/document/d/1IXWyyQjQyK29UQxj4MnQIxBLKNQB6kxWG_tGoSa4-1Y/edit

Wendy is going to take a look at the copy and give me some edits later today.

I have a few questions for @ghing about the emails:
- In the published story email, it would be nice to have a preview that looks similar to how stories are displayed in the homepage slider (title, byline, thumbnail, truncated summary, link to full story and link to other Floodlight stories). Is this doable?
- I also considered having the embed code in the email, as text, so the user can directly copy and paste from there. Thoughts on that?
- The final feature on the published story email was a list of **other** Floodlight stories that have been recently published, to encourage users to participate in the site by reading stories. Would it be doable to have a list of, say, the three most recently published stories in the email?

If those aren't feasible, we can have a simple standard text email instead.

**Comment by ghing** [comment]
> In the published story email, it would be nice to have a preview that
> looks similar to how stories are displayed in the homepage slider (title,
> byline, thumbnail, truncated summary, link to full story and link to other
> Floodlight stories). Is this doable?

It's doable, but the big consideration is the amount of testing this might
take to make sure this looks good across different platforms, devices and
web-based email providers/browsers.

In any case, I think we'll want to have a text version as a fallback or for
people who prefer text.

> I also considered having the embed code in the email, as text, so the
> user can directly copy and paste from there. Thoughts on that?

There's no technical roadblock to including this. Off the top of my head,
I'm a little concerned about how different mail clients handle linebreaks.
I'm also worried about just including too much information. We could try
this and just axe it if it turns out to be problematic.

> The final feature on the published story email was a list of other
> Floodlight stories that have been recently published, to encourage users to
> participate in the site by reading stories. Would it be doable to have a
> list of, say, the three most recently published stories in the email?

We can do this.

I thought @bpawlak also mentioned including a list of all of a user's
unpublished stories. Do we still want to do this?

**Comment by jwirfs-brock** [comment]
@ghing 
I'll get to work on the text-only backup -- that definitely makes sense. Any suggestions on the simplest way to come up with a preview format that looks good on different platforms?

Yup, I am also going back and forth about whether to include the embed code. I was also thinking we could just mention that the embed code is available and then link to the story detail page.

Good catch on the list of unpublished stories. I'll add that to the unpublished story email copy.

**Comment by ghing** [comment]
> Any suggestions on the simplest way to come up with a preview format that looks good on different platforms?

I'll have to look up best practices for HTML e-mail. This is an area where I don't have a lot of experience. One piece of advice I've heard over the years (which isn't really a good fit for this case), is "use lightly modified templates from an existing mass-email provider".  However, looking at the HTML of e-mails that you think work well is probably a good start. If you could start collecting some for me to look at, that would be awesome.

One thing that I particularly hate, especially because I read e-mail a lot on my phone, is multiple columns.  Personally, I'd try to keep e-mails to a single column of information and use the HTML less for layout and more to add hierarchy or draw attention to certain information.  Some sketches of what you're imagining would be helpful to me here.

**Comment by ghing** [comment]
Note that it's easy to [send HTML email in Django](https://docs.djangoproject.com/en/1.4/topics/email/#sending-alternative-content-types). The hard part is getting the template right.

@jwirfs-brock, a few questions:
- How do we want opt-out to work for these messages?
- Do we want to spend the time to implement HTML e-mails in this issue, or just do plain text and see if people use this feature and then implement HTML later?

**Comment by ghing** [comment]
[How to Code HTML Emails](http://kb.mailchimp.com/article/how-to-code-html-emails) from the MailChimp Knowledge Base is a helpful resource for designing HTML templates for email

**Comment by jwirfs-brock** [comment]
@ghing 
- How do we want opt-out to work for these messages?

We have a feature in the user-settings that lets users opt in or out of communications. We should hook that up to this.

Here are the options in the "communications preferences" section of the user account right now (http://floodlightproject.org/en/accounts/notifications/):

**Floodlight updates**
Email me with...
- Updates about new Floodlight features, events and storytelling tips
- Administrative account updates
- A monthly digest of featured Floodlight stories

**Notifications about my stories**
Email me when...
- One of my stories is featured on the Floodlight homepage or newsletter
- Someone comments on one of my stories

We need to fix this list so that it actually matches up with our e-mail alerts (weekly newsletter, updates about your stories, administrative account updates). Should I open a separate issue for this?
- Do we want to spend the time to implement HTML e-mails in this issue, or just do plain text and see if people use this feature and then implement HTML later?

Would we be able to include the story's thumbnail and summary without HTML? If so, let's just go text only and implement the HTML later.

**Comment by ghing** [comment]
@jwirfs-brock:

> We need to fix this list so that it actually matches up with our e-mail alerts (weekly newsletter, updates about your stories, administrative account updates). Should I open a separate issue for this?

Please open a separate issue for this and put the copy updates there.

> Would we be able to include the story's thumbnail and summary without HTML? 

We could put the story summary text in a text-only e-mail.  We could link to the featured image, but it wouldn't show up inline (unless some clients do some guessing and handle this automagically)

**Comment by jwirfs-brock** [comment]
@ghing The email copy is now finalized (see links above). The unpublished story email is text-only. The published story email should be HTML, but I included a text-only version as a backup.

What do you need from me in order to create the HTML template? Would a sketch be sufficient?

**Comment by ghing** [comment]
For the HTML e-mail template a sketch would be good. I might have some follow-up questions for you about it and there might be some limitations in terms of layout and styling for things to be as compatible as possible with different mail clients.

**Comment by bpawlak** [comment]
Campaign Monitor offers a lot of great tips, including:
- Make it scan-friendly - Limit body copy to easily-readable paragraphs, preferably under 60 characters in width.
- Avoid using mailto links in HTML email
- Align to the left
- Adding bulletproof background images and buttons to your email newsletters
- Using web fonts in email

http://www.campaignmonitor.com/blog/tips-resources

**Comment by ghing** [comment]
@jwirfs-brock I've been testing the HTML email template using [Litmus](https://litmus.com/) and it's generally looks pretty good as we created such a simple template. One minor issue is that, in Android's mail client, the right-hand-side of the content gets cut off:

![screenshot from 2013-07-22 12 24 41](https://f.cloud.github.com/assets/110420/836620/cb30e944-f2f4-11e2-9236-ac46a88555f0.png)

I use the Android mail client on my phone and this behavior happens with a lot of HTML emails I get, so I don't think it's too unexpected for users. However, if you want to try to avoid this, we could move the story titles above the images instead of to the right of them.  Let me know if the current display is acceptable, or if you want me to try to reconfigure it.

**Comment by jwirfs-brock** [comment]
Yeah, let's fix it in the name of good practices. I'd recommend putting the
story title above the image and the "view story" link below. Did you have a
different approach?

**Comment by ghing** [comment]
@jwirfs-brock 

My thought was the same as yours: 

```
Title
Image
Links
```

But what about the byline and summary?

**Comment by jwirfs-brock** [comment]
We could do:
Title
Byline
Image
Summary
Links

Then it would roughly mirror our embed widget, right? Alternatively, we
could do:

Image
Title
Byline
Summary
Links

Maybe play with it and see what looks best in an email?

**Comment by jwirfs-brock** [comment]
Also, consider that a some email clients (like Gmail) strip out the images
until you enable then. Given that, what looks best?

**Comment by bpawlak** [comment]
I think the title (which is also a link) should be first. We want the email to be immediately actionable and not require people to have to think too much.

**Comment by jwirfs-brock** [comment]
Good point, Bill.
